src/stdune/interned.ml

```diff
@@ -70,6 +70,9 @@ module type Settings = sig
 end
 
 module Make (R : Settings) () = struct
+  (* The mutable tables in this module can be made safe if we stop leaking
+     information about the string representation, e.g. by switching from [int]s
+     to an opaque data type. *)
```

Comment:
Where do we leak information about the string representation? The type that this functor produces is abstract.

Reply:
I update the comment: we expose `compare` that does leak information.